Tests/CodegenCLITests/VersionCheckerTests.swift

```diff
@@ -0,0 +1,231 @@
+    // given
+    let apolloVersion = "1.0.0.test-1"
+    try fileManager.createFile(
+      body: version1PackageResolvedFileBody(apolloVersion: apolloVersion),
+      named: "Package.resolved"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: fileManager.directoryURL
+    )
+
+    // then
+    expect(result).to(equal(.versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)))
+  }
+
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInProjectRoot_withVersion2FileFormat_hasMatchingVersion_returns_versionMatch() throws {
+    // given
+    try fileManager.createFile(
+      body: version2PackageResolvedFileBody(apolloVersion: Constants.CLIVersion),
+      named: "Package.resolved"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: fileManager.directoryURL
+    )
+
+    // then
+    expect(result).to(equal(.versionMatch))
+  }
+
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInProjectRoot_withVersion2FileFormat_hasNonMatchingVersion_returns_versionMismatch() throws {
+    // given
+    let apolloVersion = "1.0.0.test-1"
+    try fileManager.createFile(
+      body: version2PackageResolvedFileBody(apolloVersion: apolloVersion),
+      named: "Package.resolved"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: fileManager.directoryURL
+    )
+
+    // then
+    expect(result).to(equal(.versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)))
+  }
+
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInProjectRoot_noProjectRootURL_hasMatchingVersion_returns_versionMatch() throws {
+    // given
+    try fileManager.createFile(
+      body: version2PackageResolvedFileBody(apolloVersion: Constants.CLIVersion),
+      named: "Package.resolved"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: nil
+    )
+
+    // then
+    expect(result).to(equal(.versionMatch))
+  }
+
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInProjectRoot_noProjectRootURL_hasNonMatchingVersion_returns_versionMismatch() throws {
+    // given
+    let apolloVersion = "1.0.0.test-1"
+    try fileManager.createFile(
+      body: version2PackageResolvedFileBody(apolloVersion: apolloVersion),
+      named: "Package.resolved"
+    )
+
+    // when
+    let result = try VersionChecker.matchCLIVersionToApolloVersion(
+      projectRootURL: nil
+    )
+
+    // then
+    expect(result).to(equal(.versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)))
+  }
+
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeProject_withVersion2FileFormat_hasMatchingVersion_returns_versionMatch() throws {
```

Comment:
Is only using version 2 of the package file format linked to the change in swift tools version of 5.6? There is a chance that a project doesn't regenerate their schema package and still uses an older version of Xcode; do we know how far back it began to use v2 of the package format?

Reply:
Yeah v2 is linked to the swift tools version 5.6.

Yes, the Version Checker does support both v1 and v2. There are tests that it parses a v1 package. But there is no reason to add another test for that here. We have tests for the behavior, and writing a new test that is never going to fail and doesn't require any new code to be written to pass violates TDD principles and is honestly a waste of time. Unless there is actually some new behavior to test, having multiple tests that test the same lines of code is not necessary.